internal/querycoordv2/meta/target_manager.go

```diff
@@ -214,17 +213,22 @@ func (mgr *TargetManager) PullNextTargetV2(broker Broker, collectionID int64, ch
 
 		for _, infos := range channelInfos {
 			merged := mgr.mergeDmChannelInfo(infos)
-			dmChannels[merged.GetChannelName()] = merged
+			channels[merged.GetChannelName()] = merged
 		}
 		return nil
 	}
 
 	err := retry.Do(context.TODO(), getRecoveryInfo, retry.Attempts(10))
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	return segments, dmChannels, nil
+	indexes, err := broker.DescribeIndex(context.Background(), collectionID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
shall we put it into retry scope?

Reply:
no need to retry manually, the observer would try this constantly